js_hw6/task_02.js

```diff
@@ -0,0 +1,267 @@
+
+function addDepartment(id, name) {
+    const deppartment = generateNewDepartment(name);
+    const enterprise = findEnterprise(id);
+    enterprise.departments.push(deppartment);  
+};
+
+function findEnterprise(id) {
+    const enterprise = enterprises.find((enterprise) => enterprise.id === id);
+    if(!enterprise) throw new Error("Enterprise not found");
+    return enterprise;
+};
+function generateNewDepartment(name) {
+    const newDepartment = {
+        id: getNewId(),
+        name: name,
+        employees_count: 0,
+      }
+    return newDepartment;
+};
+
+addDepartment(9, "Новый отдел");
+console.log(JSON.stringify(enterprises));
+
+// 5. Написать функцию для редактирования названия предприятия. Принимает в качестве аргумента id предприятия и новое имя предприятия.
+
+// Пример:
+// editEnterprise(1, "Новое название предприятия")
+
+function editEnterprise(id, name) {
+    const enterprise = findEnterprise(id);
+    enterprise.name = name;
+};
+
+editEnterprise(5, "New name");
+console.log(enterprises);
+
+// 6. Написать функцию для редактирования названия отдела. Принимает в качестве аргумента id отдела и новое имя отдела.
+
+// Пример:
+// editDepartment(7, "Новое название отдела")
+
+function editDepartment(id, name) {
+    const department = findDepartment(id);
+    department.name = name;
+};
+
+function findDepartment(id) {
+    let result;
+    for (const enterprise of enterprises) {        
+        result = enterprise.departments.find((department) => department.id === id);
+        if (result) break;    
+    }
+    if (!result) throw new Error("No such department");
+    return result;
+};
+        
+
+editDepartment(2, 'Новый отдел тестирования');
+console.log(JSON.stringify(enterprises));
+
+// 7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+
+// Пример:
+// deleteEnterprise(1)
+
+function deleteEnnterpriseById(id) {
+    let  isDeleted = false;
+    for (const enterprise of enterprises) {
+        if (enterprise.id === id) {
+            enterprises.splice(enterprises.indexOf(enterprise), 1);
+            isDeleted = true;
+            break;
+        }
+    };
+    if (!isDeleted) throw new Error(`Enterprise with id = ${id} not found`);
+}
+
+deleteEnnterpriseById(4);
+console.log(enterprises);
+
+// 8. Написать функцию для удаления отдела. В качестве аргумента принимает id отдела. Удалить отдел можно только, если в нем нет сотрудников.
+
+// Пример:
+// deleteDepartment(3)
+
+function deleteDepartment(id) {
+    const department = findDepartment(id);    
+    if (department.employees_count === 0) throw new Error (`Department with id ${id} has no employees`);
+    for (const enterprise of enterprises) {
+        for (const department of enterprise.departments) {
+            if (department.id === id && department.employees_count > 0) {
+                enterprise.departments.splice(enterprise.departments.indexOf(department), 1);
+            }
+        }
+    }
```

Comment:
You already checked whether the department has an employee
 ```suggestion
                let index = enterprise.departments.findIndex((el) => department.id === el.id);
                enterprise.departments.splice(index, 1);
    }
```

Reply:
Good catch, thanks